src/features/common/data/state-cleanup.ts

```diff
@@ -32,13 +33,14 @@ const stateCleanupEffect = atomEffect((get, set) => {
     removeExpired(applicationResultsAtom)
     removeExpired(assetMetadataResultsAtom)
     removeExpired(assetResultsAtom)
+    removeExpired(applicationsAppSpecsAtom)
```

Comment:
Assume you've tested that once an app spec has been evicted, a fetch to the DB occurs again?